model/riscv_sys_regs.sail

```diff
@@ -824,3 +827,88 @@ function read_seed_csr() -> xlenbits = {
 
 /* Writes to the seed CSR are ignored */
 function write_seed_csr () -> option(xlenbits) = None()
+
+/* vector csrs */
+register vstart : bits(16) /* use the largest possible length of vstart */
+register vxsat  : bits(1)
+register vxrm   : bits(2)
+register vl     : xlenbits
+register vlenb  : xlenbits
+
+bitfield Vtype  : xlenbits = {
+  vill      : xlen - 1,
+  reserved  : xlen - 2 .. 8,
+  vma       : 7,
+  vta       : 6,
+  vsew      : 5 .. 3,
+  vlmul     : 2 .. 0
+}
+register vtype : Vtype
+
+/* the dynamic selected element width (SEW) */
+/* this returns the power of 2 for SEW */
+val get_sew_pow : unit -> {|3, 4, 5, 6|} effect {escape, rreg}
+function get_sew_pow() = {
+  let SEW_pow : {|3, 4, 5, 6|} = match vtype.vsew() {
+    0b000 => 3,
+    0b001 => 4,
+    0b010 => 5,
+    0b011 => 6,
+    _     => {assert(false, "invalid vsew field in vtype"); 0}
+  };
+  SEW_pow
+}
+/* this returns the actual value of SEW */
+val get_sew : unit -> {|8, 16, 32, 64|} effect {escape, rreg}
+function get_sew() = {
+  match get_sew_pow() {
+    3 => 8,
+    4 => 16,
+    5 => 32,
+    6 => 64
+  }
+}
+/* this returns the value of SEW in bytes */
+val get_sew_bytes : unit -> {|1, 2, 4, 8|} effect {escape, rreg}
+function get_sew_bytes() = {
+  match get_sew_pow() {
+    3 => 1,
+    4 => 2,
+    5 => 4,
+    6 => 8
+  }
+}
+
+/* the vector register group multiplier (LMUL) */
+/* this returns the power of 2 for LMUL */
+val get_lmul_pow : unit -> {|-3, -2, -1, 0, 1, 2, 3|} effect {escape, rreg}
+function get_lmul_pow() = {
+  match vtype.vlmul() {
+    0b101 => -3,
+    0b110 => -2,
+    0b111 => -1,
+    0b000 => 0,
+    0b001 => 1,
+    0b010 => 2,
+    0b011 => 3,
+    _     => {assert(false, "invalid vlmul field in vtype"); 0}
+  }
+}
+
+enum agtype = { UNDISTURBED, AGNOSTIC }
+
+val get_vtype_vma : unit -> agtype effect {rreg}
+function get_vtype_vma() = {
+  match vtype.vma() {
+    0b0 => UNDISTURBED,
+    0b1 => AGNOSTIC
+  }
+}
+
+val get_vtype_vta : unit -> agtype effect {rreg}
+function get_vtype_vta() = {
+  match vtype.vta() {
+    0b0 => UNDISTURBED,
+    0b1 => AGNOSTIC
```

Comment:
You want a function that does bits(1) -> agtype and use that for both of these rather than duplicate

Reply:
modified as suggested